src/txorphanage.cpp

```diff
@@ -15,14 +14,35 @@ static constexpr int64_t ORPHAN_TX_EXPIRE_TIME = 20 * 60;
 /** Minimum time between orphan transactions expire time checks in seconds */
 static constexpr int64_t ORPHAN_TX_EXPIRE_INTERVAL = 5 * 60;
 
+void TxOrphanage::SubtractOrphanBytes(unsigned int size, NodeId peer)
+{
+    AssertLockHeld(m_mutex);
+    Assume(m_peer_bytes_used.count(peer) > 0);
+    Assume(m_peer_bytes_used.at(peer) >= size);
```

Comment:
Not a big fan of using `Assume` like this because we can't test these conditions but they **can** happen in production (if the caller is doing something wrong).

I'd suggest just returning if these assumptions don't hold.